### PR TITLE
Deploy injective and inEVM

### PR DESCRIPTION
**Pull request by yorhodes**
### Description

- Deploy core to injective and inEVM networks

### Drive-by changes

- Update cw-hyperlane dependency to https://github.com/hyperlane-xyz/cosmwasm/releases/tag/v0.0.6

### Related issues

- 

### Backward compatibility

Yes

### Testing

- Fork tests
- e2e tests

### CODE REVIEW

File: typescript/sdk/src/consts/chainMetadata.ts [hunk 1]
```diff
@@ -415,6 +415,55 @@ export const gnosis: ChainMetadata = {
...
+  blockExplorers: [],
+  blocks: {
+    confirmations: 1,
+    estimateBlockTime: 3,
+    reorgPeriod: 1,
+  },
+  chainId: 'injective-1',
+  domainId: 6909546,
+  displayName: 'Injective',
+  displayNameShort: 'Injective',
```
Comment (jmrossy) [inline typescript/sdk/src/consts/chainMetadata.ts:456]: Short is optional if the name is already pretty short

File: typescript/sdk/src/consts/chains.ts [hunk 2]
```diff
@@ -68,6 +70,7 @@ export const Mainnets: Array<CoreChainName> = [
   Chains.base,
   Chains.scroll,
   Chains.polygonzkevm,
+  Chains.injective,
```
Comment (jmrossy) [inline typescript/sdk/src/consts/chains.ts:73]: is inEVM missing here?

File: typescript/sdk/src/consts/chainMetadata.ts [hunk 3]
```diff
@@ -415,6 +415,55 @@ export const gnosis: ChainMetadata = {
...
+  name: Chains.injective,
+  nativeToken: {
+    decimals: 6,
+    name: 'Injective',
+    symbol: 'INJ',
+  },
+  protocol: ProtocolType.Cosmos,
+  rpcUrls: [{ http: 'https://sentry.tm.injective.network:443' }],
```
Comment (jmrossy) [inline typescript/sdk/src/consts/chainMetadata.ts:464]: Chains chains also need a `restURLs` field for the warp UI to work right (see neutron)

File: typescript/infra/scripts/deploy.ts [hunk 4]
```diff
@@ -69,45 +69,44 @@ async function main() {
...
     const ismFactory = HyperlaneIsmFactory.fromAddressesMap(
       getAddresses(environment, Modules.PROXY_FACTORY),
       multiProvider,
     );
     deployer = new HyperlaneCoreDeployer(multiProvider, ismFactory);
   } else if (module === Modules.WARP) {
     const owner = deployerAddress;
-    const neutronRouter =
-      '6b04c49fcfd98bc4ea9c05cd5790462a39537c00028333474aebe6ddf20b73a3';
+    const injectiveRouter =
```
Comment (jmrossy) [inline typescript/infra/scripts/deploy.ts:83]: Strange to see an address hardcoded in here

File: typescript/sdk/src/consts/chainMetadata.ts [hunk 5]
```diff
@@ -415,6 +415,60 @@ export const gnosis: ChainMetadata = {
   ],
 };
 
+export const inevm: ChainMetadata = {
+  blockExplorers: [
+    {
+      apiUrl: 'https://inevm.calderaexplorer.xyz/api',
+      family: ExplorerFamily.Blockscout,
+      name: 'Caldera inEVM Explorer',
+      url: 'https://inevm.calderaexplorer.xyz/',
```
Comment (jmrossy) [inline typescript/sdk/src/consts/chainMetadata.ts:424]: Avoid tailing slashes please for consistency

File: typescript/sdk/src/consts/chainMetadata.ts [hunk 6]
```diff
@@ -415,6 +415,60 @@ export const gnosis: ChainMetadata = {
...
+    decimals: 18,
+    name: 'Injective',
+    symbol: 'INJ',
+  },
+  protocol: ProtocolType.Cosmos,
+  slip44: 118,
+  bech32Prefix: 'inj',
+  // temporarily testnet
+  grpcUrls: [{ http: 'testnet.sentry.chain.grpc.injective.network:443' }],
+  rpcUrls: [{ http: 'https://k8s.testnet.tm.injective.network/' }],
```
Comment (jmrossy) [inline typescript/sdk/src/consts/chainMetadata.ts:468]: ditto

File: typescript/sdk/src/consts/environments/mainnet.json [hunk 7]
```diff
@@ -212,5 +212,22 @@
...
+    "domainRoutingIsm": "0xBD70Ea9D599a0FC8158B026797177773C3445730",
+    "proxyAdmin": "0x0761b0827849abbf7b0cC09CE14e1C93D87f5004",
+    "storageGasOracle": "0x6119E37Bd66406A1Db74920aC79C15fB8411Ba76",
+    "interchainGasPaymaster": "0x19dc38aeae620380430C200a6E990D5Af5480117",
+    "merkleTreeHook": "0x0972954923a1e2b2aAb04Fa0c4a0797e5989Cd65",
+    "aggregationHook": "0xe0dDb5dE7D52918237cC1Ae131F29dcAbcb0F62B",
+    "protocolFee": "0x0D63128D887159d63De29497dfa45AFc7C699AE4",
+    "mailbox": "0x2f2aFaE1139Ce54feFC03593FeE8AB2aDF4a85A7",
+    "validatorAnnounce": "0x15ab173bDB6832f9b64276bA128659b0eD77730B",
+    "interchainSecurityModule": "0xBD70Ea9D599a0FC8158B026797177773C3445730"
```
Comment (jmrossy) [inline typescript/sdk/src/consts/environments/mainnet.json:231]: testRecipient & testTokenRecipient?
Looks like they're missing for manta above too

File: typescript/sdk/src/metadata/chainMetadataTypes.ts [hunk 8]
```diff
@@ -245,6 +245,7 @@ export type BlockExplorer = Exclude<
 >[number];
 
 export function isValidChainMetadata(c: ChainMetadata): boolean {
+  ChainMetadataSchema.parse(c);
```
Comment (jmrossy) [inline typescript/sdk/src/metadata/chainMetadataTypes.ts:248]: `is*` functions like this shouldn't throw IMO, or the return value is never false

File: typescript/sdk/src/core/MultiProtocolCore.ts [hunk 9]
```diff
@@ -37,6 +37,7 @@ export class MultiProtocolCore extends MultiProtocolApp<
     if (!envAddresses) {
       throw new Error(`No addresses found for ${env}`);
     }
+    // @ts-ignore
```
Comment (jmrossy) [inline typescript/sdk/src/core/MultiProtocolCore.ts:40]: Y dis?

File: typescript/sdk/src/consts/chainMetadata.ts (hunk 6 above)
Comment (jmrossy) [reply typescript/sdk/src/consts/chainMetadata.ts:468]: Also I recommend adding more RPCs to take advantage of SmartProvider's fallback

File: typescript/sdk/src/consts/chainMetadata.ts [hunk 10]
```diff
@@ -415,6 +415,60 @@ export const gnosis: ChainMetadata = {
...
+  displayName: 'Injective EVM',
+  displayNameShort: 'inEVM',
+  name: Chains.inevm,
+  nativeToken: {
+    decimals: 18,
+    name: 'Injective',
+    symbol: 'INJ',
+  },
+  protocol: ProtocolType.Ethereum,
+  rpcUrls: [{ http: 'https://inevm.calderachain.xyz/http' }],
```
Comment (jmrossy) [inline typescript/sdk/src/consts/chainMetadata.ts:443]: I recommend adding more RPCs to take advantage of SmartProvider's fallback:
https://github.com/cosmos/chain-registry/blob/master/injective/chain.json#L217

File: rust/config/mainnet3_config.json [hunk 11]
```diff
@@ -860,4 +860,4 @@
     }
   },
   "defaultRpcConsensusType": "fallback"
-}
+}
```
Comment (nambrot) [inline rust/config/mainnet3_config.json:863]: was this intentional?

File: rust/utils/run-locally/Cargo.toml [hunk 12]
```diff
@@ -33,4 +33,4 @@ hpl-interface.workspace = true
 cosmwasm-schema.workspace = true
 
 [features]
-cosmos = []
\ No newline at end of file
+cosmos = []
```
Comment (nambrot) [inline rust/utils/run-locally/Cargo.toml:36]: ditto

File: rust/utils/run-locally/src/cosmos/mod.rs [hunk 13]
```diff
@@ -57,8 +57,8 @@ fn default_keys<'a>() -> [(&'a str, &'a str); 6] {
     ]
 }
 
-const CW_HYPERLANE_GIT: &str = "https://github.com/many-things/cw-hyperlane";
-const CW_HYPERLANE_VERSION: &str = "0.0.6-rc6";
+const CW_HYPERLANE_GIT: &str = "https://github.com/hyperlane-xyz/cosmwasm";
+const CW_HYPERLANE_VERSION: &str = "injective-testnet";
```
Comment (nambrot) [inline rust/utils/run-locally/src/cosmos/mod.rs:61]: Is that our versioning scheme now?

File: typescript/infra/config/environments/testnet4/validators.ts [hunk 14]
```diff
@@ -289,6 +289,18 @@ export const validatorChainConfig = (
         'polygonzkevmtestnet',
       ),
     },
+    injective: {
```
Comment (nambrot) [inline typescript/infra/config/environments/testnet4/validators.ts:292]: Stale?

File: typescript/infra/config/environments/testnet4/warp/addresses.json [hunk 15]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (nambrot) [inline typescript/infra/config/environments/testnet4/warp/addresses.json:1]: Should this be added?

**Review by nambrot — APPROVED** [review]

File: rust/utils/run-locally/src/cosmos/mod.rs (hunk 13 above)
Comment (yorhodes) [reply rust/utils/run-locally/src/cosmos/mod.rs:61]: idk what the other one was :P
this made it easier to follow

File: typescript/infra/config/environments/testnet4/validators.ts (hunk 14 above)
Comment (yorhodes) [reply typescript/infra/config/environments/testnet4/validators.ts:292]: will wait until @tkporter 's PR gets merged

**Pull request merged**